core/storage/question/gae_models_test.py

```diff
@@ -199,3 +199,43 @@ def test_get_models_by_question_id(self):
                 'question_id3')
         )
         self.assertEqual(len(question_skill_links), 0)
+
+    def test_get_question_skill_links_with_constant_number_per_skill(self):
+        questionskilllink_model1 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id1', 'skill_id1', 0.1)
+            )
+        questionskilllink_model2 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id2', 'skill_id1', 0.5)
+            )
+        questionskilllink_model3 = (
+            question_models.QuestionSkillLinkModel.create(
+                'question_id3', 'skill_id2', 0.8)
+            )
+
+        question_models.QuestionSkillLinkModel.put_multi_question_skill_links(
+            [questionskilllink_model1, questionskilllink_model2,
+             questionskilllink_model3])
+
+        question_skill_links = (
+            question_models.QuestionSkillLinkModel.
+            get_question_skill_links_with_constant_number_per_skill(
+                2, ['skill_id1', 'skill_id2']
+            )
+        )
+        self.assertEqual(len(question_skill_links), 2)
+        self.assertEqual(question_skill_links[0].skill_id, 'skill_id1')
+        self.assertEqual(question_skill_links[1].skill_id, 'skill_id2')
+
+        # Testing for queries that retrieve more questions than available.
```

Comment:
Use separate tests here, explaining what the behaviour being tested is in each case.

Add additional tests to handle other cases. See comments in main file. Think of other "failure"/problematic cases too.

Reply:
Done.